extensions/hibernate-orm/deployment/src/main/java/io/quarkus/hibernate/orm/deployment/HibernateOrmProcessor.java

```diff
@@ -905,6 +908,22 @@ public void produceLoggingCategories(BuildProducer<LogCategoryBuildItem> categor
         }
     }
 
+    @BuildStep(onlyIf = NativeBuild.class)
+    public void test(CombinedIndexBuildItem index,
+            BuildProducer<ReflectiveClassBuildItem> reflective) {
+        Collection<AnnotationInstance> annotationInstance = index.getIndex().getAnnotations(STATIC_METAMODEL);
+        if (annotationInstance != null) {
+
+            String[] metamodel = annotationInstance.stream()
+                    .map(a -> a.target().asClass().toString())
+                    .toArray(String[]::new);
+
+            if (metamodel.length > 0) {
```

Comment:
Thinking about it a bit more, just test if the collection is empty above instead of null and drop that check.

Reply:
Done